### DISCUSSION TITLE
Adding S3 upload provider to Strapi Docker image - Stuck

### DISCUSSION THREAD

**Discussion by get-data-** [strapi/strapi#6194] [Legacy Topics]
I have a docker-compose file which uses Strapi and Nginx which I have used without any issues. I would like to add the `strapi-provider-upload-aws-s3` plugin. I am stuck on your custom example and at what point I'd apply this plugin change. Should this be an extra step in my Dockerfile post yarn install? Does this require me to run the `yarn build --clean` after I complete the `yarn strapi new .` command?

I'm looking at the following examples as guidance:
**custom dockerfile:**
https://github.com/strapi/strapi-docker/tree/master/examples/custom
**docker-compose example using Volume which maps app data:**
https://github.com/strapi/strapi-docker/blob/master/examples/postgresql/docker-compose.yml

Any help or guidance would be greatly appreciated.

**Comment by lauriejim** [comment]
Hello! What is the version of Strapi you are using?
If it's after the `beta.20` - you will have to check this configuration here - https://strapi.io/documentation/3.0.0-beta.x/plugins/upload.html#using-a-provider
Add the node module in your application, add your config file.
Then it should work.
